presto-orc/src/main/java/io/prestosql/orc/reader/BooleanStreamReader.java

```diff
@@ -103,10 +103,16 @@ public Block readBlock(Type type)
             }
             dataStream.getSetBits(type, nextBatchSize, builder);
         }
+        else if (dataStream == null) {
```

Comment:
I would move this clause to be above the declaration of the builder variable.  Then, I would construct a block builder directly with a size of `1`, which will avoid over allocation.  Something like:
```java
if (dataStream == null) {
    presentStream.skip(nextBatchSize);
    readOffset = 0;
    nextBatchSize = 0;
    return new RunLengthEncodedBlock(type.createBlockBuilder(null, 1).appendNull().build(), nextBatchSize);
}